frontend/public/components/marketplace/kubernetes-marketplace.jsx

```diff
@@ -0,0 +1,132 @@
+
+const normalizePackageManifests = (packageManifests, kind) => {
+  const activePackageManifests = _.filter(packageManifests, packageManifest => {
+    return !packageManifest.status.removedFromBrokerCatalog;
+  });
+  return _.map(activePackageManifests, packageManifest => {
+    const tileName = packageManifest.metadata.name;
+    const iconClass = 'fa fa-clone'; // TODO: get this info from the packagemanifest
+    const iconObj = _.get(packageManifest, 'status.channels[0].currentCSVDesc.icon[0]', null);
+    const tileImgUrl = iconObj ? `data:${_.get(iconObj, 'mediatype')};base64,${_.get(iconObj, 'base64data')}` : null;
+    const tileIconClass = tileImgUrl ? null : iconClass;
+    const tileDescription = packageManifest.metadata.description;
+    const tileProvider = packageManifest.metadata.labels.provider;
+    const tags = packageManifest.metadata.tags;
+    return {
+      obj: packageManifest,
+      kind,
+      tileName,
+      tileIconClass,
+      tileImgUrl,
+      tileDescription,
+      tileProvider,
+      tags,
+    };
+  });
+};
+
+const getItems = (props) => {
+  const {packagemanifests, loaded} = props;
+  let packageManifestItems = null;
+  if (!loaded) {
+    return [];
+  }
+  if (packagemanifests) {
+    packageManifestItems = normalizePackageManifests(packagemanifests.data, 'PackageManifest');
+  }
+  return _.sortBy([...packageManifestItems], 'tileName');
+};
+
+class MarketplaceListPage extends React.Component {
+  state = {};
+
+  static getDerivedStateFromProps(props, state) {
+    const {packagemanifests} = props;
+    if (packagemanifests !== _.get(state, 'packagemanifests')) {
+      const items = getItems(props);
+      return {items, packagemanifests};
+    }
+  }
+
+  renderTiles() {
+    const {items} = this.state;
+
+    return (
+      <CatalogTileView.Category totalItems={items.length} viewAll={true}>
+        {_.map(items, ((item) => {
+          const {tileName, tileImgUrl, tileIconClass, tileProvider, tileDescription} = item;
+          const uid = tileName;
+          const iconClass = tileIconClass ? `icon ${normalizeIconClass(tileIconClass)}` : null;
+          const vendor = tileProvider ? `Provided by ${tileProvider}` : null;
+          return <CatalogTile
+            id={uid}
+            key={uid}
+            title={tileName}
+            iconImg={tileImgUrl}
+            iconClass={iconClass}
+            vendor={vendor}
+            description={tileDescription}
+          />;
+        }))}
+      </CatalogTileView.Category>
+    );
+  }
+
+  render() {
+    const {loaded, loadError} = this.props;
+    const {items} = this.state;
+    return <StatusBox data={items} loaded={loaded} loadError={loadError} label="Resources">
+      <div className="co-catalog-page">
+        <div className="co-catalog-page__content">
+          <div>
+            <div className="co-catalog-page__num-items">{_.size(items)} items</div>
+          </div>
+          <CatalogTileView>
+            {this.renderTiles()}
```

Comment:
I may have missed it, but why is this a method and not just inline JSX?

Reply:
it may be calling other methods later on that have to do with filtering and categorizing so we thought it better to use a method for now